client/core/trade.go

```diff
@@ -1612,90 +1630,107 @@ func (c *Core) redeemMatchGroup(t *trackedTrade, matches []*matchTracker, errs *
-// The redeem details are always saved even if sending the `redeem` request
-// errors or a valid ack is not received from the server. This makes it
-// possible to resend the `redeem` request at a later time.
+// sendInitAsync starts a goroutine to send an `redeem` request for the specified
+// match and save the server's ack sig to db. Sends a notification if an error
+// occurs while sending the request or validating the server's response.
 //
 // This method modifies match fields and MUST be called with the trackedTrade
 // mutex lock held for writes.
-func (c *Core) finalizeRedeemAction(t *trackedTrade, match *matchTracker, coinID []byte) error {
-	_, _, proof, auth := match.parts()
-	if len(auth.RedeemSig) != 0 {
-		return fmt.Errorf("'redeem' already sent for match %v", match.id)
+func (c *Core) sendRedeemAsync(t *trackedTrade, match *matchTracker, coinID, secret []byte) {
+	if match.sendingRedeemAsync || match.MetaData.Proof.IsRevoked() {
+		return // no need sending `redeem` request for a revoked match
 	}
-	errs := newErrorSet("")
 
-	// Attempt to send `redeem` request and validate server ack.
-	// Not necessary for revoked matches.
-	var needsResolution bool
-	if !proof.IsRevoked() {
+	match.sendingRedeemAsync = true
+	c.log.Debugf("Sending 'redeem' request for match %s, redeem coin %v (%s)",
+		match.id, coinIDString(t.fromAssetID, coinID), t.wallets.fromAsset.Symbol)
+
+	// Send the redeem request asynchronously.
+	go func() {
+		var err error
+		defer func() {
+			t.mtx.Lock()
+			match.sendingRedeemAsync = false
+			if err != nil {
+				corder := t.coreOrderInternal()
+				details := fmt.Sprintf("Error notifying DEX of redemption for match %s: %v", match.id, err)
+				t.notify(newOrderNote(SubjectSwapError, details, db.ErrorLevel, corder))
+			}
+			t.mtx.Unlock()
+		}()
+
 		msgRedeem := &msgjson.Redeem{
 			OrderID: t.ID().Bytes(),
 			MatchID: match.id.Bytes(),
 			CoinID:  coinID,
-			Secret:  proof.Secret,
+			Secret:  secret,
 		}
 		ack := new(msgjson.Acknowledgement)
 		// The DEX may wait up to its configured broadcast timeout, but we will
 		// retry on timeout or other error.
 		timeout := t.broadcastTimeout() / 4
-		if err := t.dc.signAndRequest(msgRedeem, msgjson.RedeemRoute, ack, timeout); err != nil {
+		err = t.dc.signAndRequest(msgRedeem, msgjson.RedeemRoute, ack, timeout)
+		if err != nil {
+			err = fmt.Errorf("error sending 'redeem' message: %v", err)
 			var msgErr *msgjson.Error
-			needsResolution = errors.As(err, &msgErr) && msgErr.Code == msgjson.SettlementSequenceError
-			ack.Sig = nil // in case of partial unmarshal
-			errs.add("error sending 'redeem' message for match %s: %v", match.id, err)
-		} else if err := t.dc.acct.checkSig(msgRedeem.Serialize(), ack.Sig); err != nil {
-			ack.Sig = nil // don't record an invalid signature
-			errs.add("'redeem' ack signature error for match %s: %v", match.id, err)
+			if errors.As(err, &msgErr) && msgErr.Code == msgjson.SettlementSequenceError {
```

Comment:
Same concern about hitting this settlement sequence case without the error wrapped properly or after this.